ml-agents/mlagents/trainers/ppo/optimizer_torch.py

```diff
@@ -25,7 +26,19 @@ def __init__(self, policy: TorchPolicy, trainer_settings: TrainerSettings):
         # Create the graph here to give more granular control of the TF graph to the Optimizer.
 
         super().__init__(policy, trainer_settings)
-        params = list(self.policy.actor_critic.parameters())
+        reward_signal_configs = trainer_settings.reward_signals
+        reward_signal_names = [key.value for key, _ in reward_signal_configs.items()]
+
+        if policy.shared_critic:
+            self._critic = policy.actor
+        else:
+            self._critic = ValueNetwork(
+                reward_signal_names,
+                policy.behavior_spec.observation_specs,
+                network_settings=trainer_settings.network_settings,
+            )
+
```

Comment:
I think we need to move the critic to `default_device` like we did in the Policy